interfaces/connect.ts

```diff
@@ -0,0 +1,226 @@
+}
+
+export enum ValidityTypes {
+    VALIDITY_DAY = 'DAY',
+    VALIDITY_IOC = 'IOC',
+    VALIDITY_TTL = 'TTL',
+}
+
+export enum ExchangeTypes {
+    EXCHANGE_NSE = 'NSE',
+    EXCHANGE_BSE = 'BSE',
+    EXCHANGE_NFO = 'NFO',
+    EXCHANGE_CDS = 'CDS',
+    EXCHANGE_BCD = 'BCD',
+    EXCHANGE_BFO = 'BFO',
+    EXCHANGE_MCX = 'MCX',
+}
+
+export enum MarginTypes {
+    MARGIN_EQUITY = 'equity',
+    MARGIN_COMMODITY = 'commodity',
+}
+
+export enum StatusTypes {
+    STATUS_CANCELLED = 'CANCELLED',
+    STATUS_REJECTED = 'REJECTED',
+    STATUS_COMPLETE = 'COMPLETE',
+}
+
+export enum GTTStatusTypes {
+    GTT_TYPE_OCO = 'two-leg',
+    GTT_TYPE_SINGLE = 'single',
+    GTT_STATUS_ACTIVE = 'active',
+    GTT_STATUS_TRIGGERED = 'triggered',
+    GTT_STATUS_DISABLED = 'disabled',
+    GTT_STATUS_EXPIRED = 'expired',
+    GTT_STATUS_CANCELLED = 'cancelled',
+    GTT_STATUS_REJECTED = 'rejected',
+    GTT_STATUS_DELETED = 'deleted',
+}
+
+export enum PositionTypes {
+    POSITION_TYPE_DAY = 'day',
+    POSITION_TYPE_OVERNIGHT = 'overnight',
+}
+
+export interface KiteConnectParams {
+    /**
+     * API key issued to you.
+     */
+    api_key: string;
+    /**
+     * Token obtained after the login flow in exchange for the `request_token`.
+     * Pre-login, this will default to null, but once you have obtained it, you
+     * should persist it in a database or session to pass to the Kite Connect
+     * class initialisation for subsequent requests.
+     *
+     * Defaults to `null`
+     */
+    access_token?: string | null;
+    /**
+     * API end point root. Unless you explicitly want to send API requests to a
+     * non-default endpoint, this can be ignored.
+     *
+     * Defaults to 'https?://api.kite.trade'
+     */
+    root?: string | RegExp | Url;
+    /**
+     * Kite connect login url
+     *
+     * Defaults to 'https?://kite.trade/connect/login'
+     */
+    login_uri?: string;
+    /**
+     * If set to true, will console log requests and responses.
+     *
+     * Defaults to `false`
+     */
+    debug?: boolean;
+    /**
+     * Time (milliseconds) for which the API client will wait for a request to complete before it fails.
+     *
+     * Defaults to `7000`
+     */
+    timeout?: number;
+    /*
+     * Default url for which user will be redirected to login page
+     * defa
+     */
+    default_login_uri?: string;
+};
+
+export interface KiteConectInterface extends KiteConnectParams {
+
+};
+
+export enum ExchangeTypes {
+    NSE = 'NSE',
+    BSE = 'BPSE',
+    NFO = 'NFO',
+    BFO = 'BFO',
+    CDS = 'CDS',
+    MCX = 'MCX'
+};
+
+export enum TransactionTypes {
+    BUY = 'BUY',
+    SELL = 'SELL'
+};
+
+export enum Products {
+    NRML = 'NRML',
+    MIS = 'MIS',
+    CNC = 'CNC'
+};
+
+export enum OrderTypes {
+    LIMIT = 'LIMIT',
+    SL = 'SL',
+    SLM = 'SL-M',
+    MARKET = 'MARKET'
+};
+
+export enum Validities {
+    DAY = 'DAY',
+    IOC = 'IOC',
+};
+
+export interface OrderParams {
```

Comment:
Every property is not optional, please remove `?` from mandatory properties.

Reply:
can you tell which one is mandatory. I intetionally did them to get rid of errors!

Reply:
You can refer to jsdoc comments in the master branch (Optional properties had an `optional` attribute) or go to the current [kiteconnect docs](https://kite.trade/docs/kiteconnectjs/v3/index.html) and there you can see all the optional properties are marked optional, rest are mandatory.

e.g.
<img width="1167" alt="image" src="https://user-images.githubusercontent.com/53750093/231257412-e05b21b1-8925-435e-bf56-58208080f87c.png">